internal/backends/linuxkms/display/swdisplay/linuxfb.rs

```diff
@@ -63,24 +186,30 @@ impl LinuxFBDisplay {
             finfo
         };
 
-        let format = if vinfo.bits_per_pixel == 32 {
-            drm::buffer::DrmFourcc::Xrgb8888
-        } else if vinfo.bits_per_pixel == 16 {
-            if vinfo.red != RGB565_EXPECTED_RED_CHANNEL
-                || vinfo.green != RGB565_EXPECTED_GREEN_CHANNEL
-                || vinfo.blue != RGB565_EXPECTED_BLUE_CHANNEL
-            {
-                return Err(format!("Error using linux framebuffer: 16-bpp framebuffer does not have expected 565 format. Found red:{}/{} green:{}/{} blue:{}/{}",
-                    vinfo.red.offset, vinfo.red.length,
-                    vinfo.green.offset, vinfo.green.length,
-                    vinfo.blue.offset, vinfo.blue.length).into());
+        let format = Self::detect_pixel_format(&vinfo)
+            .map_err(|e| PlatformError::Other(format!("Error detecting pixel format: {}", e)))?;
+
+        let bpp = match vinfo.bits_per_pixel {
+            32 => 4,
+            24 => 3,
+            16 => 2,
+            _ => {
+                return Err(format!("Unsupported bits per pixel: {}", vinfo.bits_per_pixel).into());
             }
-            drm::buffer::DrmFourcc::Rgb565
-        } else {
-            return Err(format!("Error using linux framebuffer: Only 32- and 16-bpp framebuffers are supported right now, found {}", vinfo.bits_per_pixel).into());
         };
 
-        let bpp = vinfo.bits_per_pixel / 8;
+        println!("Detected framebuffer format: {:?} ({}bpp)", format, vinfo.bits_per_pixel);
+        println!(
+            "Color channels - R: {}/{}, G: {}/{}, B: {}/{}, A: {}/{}",
+            vinfo.red.offset,
+            vinfo.red.length,
+            vinfo.green.offset,
+            vinfo.green.length,
+            vinfo.blue.offset,
+            vinfo.blue.length,
+            vinfo.transp.offset,
+            vinfo.transp.length
+        );
```

Comment:
I'm not sure this should be printed unconditionally. Good for debugging, but might become annoying in production after a while :)